app/models/mentee_application_state.rb

```diff
@@ -0,0 +1,59 @@
+# == Schema Information
+#
+# Table name: mentee_application_states
+#
+#  id                         :bigint           not null, primary key
+#  note                       :text
+#  status                     :integer          default("pending"), not null
+#  created_at                 :datetime         not null
+#  updated_at                 :datetime         not null
+#  status_changed_id          :bigint
+#  user_mentee_application_id :bigint           not null
+#
+# Indexes
+#
+#  index_mentee_application_states_on_status_changed_id           (status_changed_id)
+#  index_mentee_application_states_on_user_mentee_application_id  (user_mentee_application_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (status_changed_id => users.id)
+#  fk_rails_...  (user_mentee_application_id => user_mentee_applications.id)
+#
+class MenteeApplicationState < ApplicationRecord
+  belongs_to :user_mentee_application
+
+  enum status: {
+    pending: 0,
+    stage_one: 1,
+    stage_two: 2,
+    stage_three: 3,
+    stage_four: 4,
+    accepted: 5,
+    rejected: 6
+  }
+
+  def self.next_status(mentee_application)
```

Comment:
would suggest a refactor to look more like what I wrote, though am fine if it's an instance method rather than class method.